### ISSUE TITLE
`parcel@2`'s packaged output (in `lib`) is compiled so that NODE_ENV is always "production"

### ISSUE THREAD

**Issue by timdorr** [parcel-bundler/parcel#5029]
# 🐛 bug report

I noticed that in `serve` mode, the global `NODE_ENV` was always `production` for any included tooling that reads it (namely, Babel). I checked the source here on GitHub and this appears to be the correct code: 

https://github.com/parcel-bundler/parcel/blob/771715b9e41021a4867715d94e7fc46f93d0303a/packages/core/parcel/src/cli.js#L45-L51

But if you check the built output for the npm package in the `lib` path, you'll see that is getting optimized down to always set it to `production`: https://unpkg.com/browse/parcel@2.0.0-beta.1/lib/cli.js

```js
const initialNodeEnv = process.env.NODE_ENV; // Then, override NODE_ENV to be PARCEL_BUILD_ENV (replaced with `production` in builds)
// so that dependencies of Parcel like React (which renders the cli through `ink`)
// run in the appropriate mode.

process.env.NODE_ENV = "production";
```

## 🎛 Configuration (.babelrc, package.json, cli command)

Here's an example config that will print out the current NODE_ENV

`babel.config.js`

```js
module.exports = function () {
  console.log(process.env.NODE_ENV)
}
```

This prints out `console: production`, even if I run `NODE_ENV=development parcel ...`

## 🤔 Expected Behavior

`NODE_ENV` reflects the intended environment for all tooling. 

## 😯 Current Behavior

It's getting overridden by optimized build output.

## 💁 Possible Solution

Don't do that? I'm not familiar enough with the tooling here to offer a solution, unfortunately.

**Comment by DeMoorJasper** [comment]
Why we originally did it is described in the comment above the code you mentioned where this is being set. We used to use `ink` to render the cli and this uses react which uses `NODE_ENV` to optimise code for production/dev, resulting in us running React in development mode while it should actually be production.

This is probably true for other libraries as well that's why we haven't really removed it yet, maybe we just should. We've already discussed this internally a couple days ago as well.

**Comment by mischnic** [comment]
Most plugins would read from `options.env` where the user specified NODE_ENV has the correct value (from here: https://github.com/parcel-bundler/parcel/blob/c88f810a92638c9b57ff9529076d43cc7ca8e0d5/packages/core/parcel/src/cli.js#L359-L361
) but this doesn't apply to JS configs which just use the current process.env: https://github.com/parcel-bundler/parcel/blob/c88f810a92638c9b57ff9529076d43cc7ca8e0d5/packages/core/utils/src/config.js#L87

**Comment by timdorr** [comment]
> Most plugins would read from `options.env`

For reference, the one I happen to be using that triggered this [uses process.env directly](https://github.com/emotion-js/emotion/blob/master/packages/babel-plugin-emotion/src/utils/label.js#L36).

Looking into the tooling in a bit more detail, it looks like #3383 is the ultimate culprit here. That added the inlining and DCE plugin so that Linux builds would work with babel-register. That seems a bit heavy handed IMHO, as you could have just changed the shebang to `#!/usr/bin/env -S node -r @parcel/babel-register` and that would have fixed the issue (tested that on Linux and macOS to be sure). 

But to @DeMoorJasper's point, I think removing the PARCEL_BUILD_ENV stuff entirely would probably be a good idea.

**Comment by AndrewKvalheim** [comment]
Resolved by #5611?

**Comment by timdorr** [comment]
Yep, it appears to be gone: 
https://unpkg.com/browse/parcel@2.0.0-nightly.495/lib/cli.js#L171
vs.   
https://unpkg.com/browse/parcel@2.0.0-nightly.532/lib/cli.js#L164
